tmt/export/__init__.py

```diff
@@ -0,0 +1,424 @@
+# coding: utf-8
+
+"""
+Metadata export functionality core.
+
+Internal APIs, plugin classes and shared functionality and helpers for metadata
+export of tests, plans or stories.
+"""
+
+import re
+import sys
+import traceback
+import types
+import xmlrpc.client
+from typing import (TYPE_CHECKING, Any, Callable, ClassVar, Dict, Generic,
+                    List, Optional, Tuple, Type, TypeVar, Union, cast)
+
+if sys.version_info >= (3, 8):
+    from typing import Protocol
+else:
+    from typing_extensions import Protocol
+
+import fmf
+from click import echo, style
+
+import tmt
+import tmt.utils
+
+if TYPE_CHECKING:
+    import tmt.base
+
+bugzilla: Optional[types.ModuleType] = None
+
+# Until Bugzilla gets its own annotations and recognizable imports...
+BugzillaInstance = Any
+
+
+# TODO: why this exists?
+log = fmf.utils.Logging('tmt').logger
+
+
+# For linking bugs
+BUGZILLA_XMLRPC_URL = "https://bugzilla.redhat.com/xmlrpc.cgi"
+RE_BUGZILLA_URL = r'bugzilla.redhat.com/show_bug.cgi\?id=(\d+)'
+
+
+# ignore[type-arg]: bound type vars cannot be generic, and it would create a loop anyway.
+ExportableT = TypeVar('ExportableT', bound='Exportable')  # type: ignore[type-arg]
+ExportClass = Type['ExportPlugin']
+
+_RawExportedInstance = Dict[str, Any]
+_RawExportedCollection = List[_RawExportedInstance]
+_RawExported = Union[_RawExportedInstance, _RawExportedCollection]
+
+
+# Protocols describing export methods.
+class FmfIdExporter(Protocol):
+    def __call__(self, fmf_id: 'tmt.base.FmfId') -> str:
+        pass
+
+
+class InstanceExporter(Protocol):
```

Comment:
This class does not seem to be used. Should it be removed then? Or is it present for future use?

Reply:
Correct, it was used but it's no more, export works with collections only. Dropped in 4644f9adb3b2512b0520b98ee5e18d8ccc956309.

Reply:
Good, thanks.